flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/utils/TableUtil.java

```diff
@@ -0,0 +1,429 @@
+	/**
+	 * Return a lower temp table name in lower case and use "-" as connector character.
+	 *
+	 * @return tableName
+	 */
+	public static synchronized String getTempTableName() {
+		return ("temp_" + UUID.randomUUID().toString().replaceAll("-", "_"))
+			.toLowerCase();
+	}
+
+	/**
+	 * Find the index of <code>targetCol</code> in string array <code>tableCols</code>. It will ignore the case of the
+	 * tableCols.
+	 *
+	 * @param tableCols a string array among which to find the targetCol.
+	 * @param targetCol the targetCol to find.
+	 * @return the index of the targetCol, if not found, returns -1.
+	 */
+	public static int findColIndex(String[] tableCols, String targetCol) {
+		Preconditions.checkNotNull(targetCol, "targetCol is null!");
+		for (int i = 0; i < tableCols.length; i++) {
+			if (targetCol.equalsIgnoreCase(tableCols[i])) {
+				return i;
+			}
+		}
+		return -1;
+	}
+
+	/**
+	 * Find the indices of <code>targetCols</code> in string array <code>tableCols</code>. If
+	 * <code>targetCols</code> is
+	 * null, it will be replaced by the <code>tableCols</code>
+	 *
+	 * @param tableCols  a string array among which to find the targetCols.
+	 * @param targetCols the targetCols to find.
+	 * @return the indices of the targetCols.
+	 */
+	public static int[] findColIndices(String[] tableCols, String[] targetCols) {
+		if (targetCols == null) {
+			int[] indices = new int[tableCols.length];
+			for (int i = 0; i < tableCols.length; i++) {
+				indices[i] = i;
+			}
+			return indices;
+		}
+		int[] indices = new int[targetCols.length];
+		for (int i = 0; i < indices.length; i++) {
+			indices[i] = findColIndex(tableCols, targetCols[i]);
+		}
+		return indices;
+	}
+
+	/**
+	 * Find the index of <code>targetCol</code> from the <code>tableSchema</code>.
+	 *
+	 * @param tableSchema the TableSchema among which to find the targetCol.
+	 * @param targetCol   the targetCols to find.
+	 * @return the index of the targetCol.
+	 */
+	public static int findColIndex(TableSchema tableSchema, String targetCol) {
+		return findColIndex(tableSchema.getFieldNames(), targetCol);
+	}
+
+	/**
+	 * Find the indices of <code>targetCols</code> from the <code>tableSchema</code>.
+	 *
+	 * @param tableSchema the TableSchema among which to find the targetCols.
+	 * @param targetCols  the targetCols to find.
+	 * @return the indices of the targetCols.
+	 */
+	public static int[] findColIndices(TableSchema tableSchema, String[] targetCols) {
+		return findColIndices(tableSchema.getFieldNames(), targetCols);
+	}
+
+	/**
+	 * Find the types of the <code>targetCols</code>. If the targetCol not exist, return null.
+	 *
+	 * @param tableSchema TableSchema.
+	 * @param targetCols  the targetCols to find.
+	 * @return the corresponding types.
+	 */
+	public static TypeInformation[] findColTypes(TableSchema tableSchema, String[] targetCols) {
+		TypeInformation[] types = new TypeInformation[targetCols.length];
+		for (int i = 0; i < types.length; i++) {
```

Comment:
echo previous comment - check null for `targetCols` missing

Reply:
We keep it consistent with findColIndices.